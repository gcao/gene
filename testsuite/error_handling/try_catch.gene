#!/usr/bin/env gene run
# TEST: Exception handling

# Basic try/catch
(print "basic try/catch: "
  (try
    (/ 10 2)
  catch e
    "error caught"))

# Catching division by zero
(print "division by zero: "
  (try
    (/ 10 0)
  catch e
    "caught division by zero"))

# Throwing custom error
(print "custom error: "
  (try
    (throw "Something went wrong!")
  catch e
    (.concat "caught: " e)))

# Finally block
(var cleanup_ran false)
(print "try/finally result: "
  (try
    "success"
  finally
    (set cleanup_ran true)))
(print "cleanup ran: " cleanup_ran)

# Nested try/catch
(print "nested: "
  (try
    (try
      (throw "inner error")
    catch e
      (throw (.concat "wrapped: " e)))
  catch e
    e))