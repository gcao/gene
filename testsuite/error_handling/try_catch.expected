basic try/catch: 5
division by zero: caught division by zero
custom error: caught: Something went wrong!
try/finally result: success
cleanup ran: true
nested: wrapped: inner error